stats/cloud/client.go

```diff
@@ -79,6 +82,21 @@ func (c *Client) NewRequest(method, url string, data interface{}) (*http.Request
 	return http.NewRequest(method, url, buf)
 }
 
+func (c *Client) setSessionToken(req *http.Request) error {
+	if req.Method != http.MethodPost {
+		return nil
+	}
+	if req.Header.Get(k6SessionTokenHeader) != "" {
+		return nil
+	}
```

Comment:
I don't think either of those or necessary: 
1. This can be used for all requests not only POSTs
2. I don't see why calling `setSessionToken` should not overwrite an old token (maybe the tests were failing
3. If each of 1 or 2 is not true we can just not call `setSessionToken` inside `Client.Do` but insted in `Client.CreateTestRun`

Reply:
@MStoykov Ms

> This can be used for all requests not only POSTs

Yes, but idempotency should be applied for POST only. There's nothing you should do with other requests. For k6, generating wasted HTTP header, which backend does not handle increase both bandwidth and load for our cloud API server

> I don't see why calling setSessionToken should not overwrite an old token (maybe the tests were failing

The point is that if the caller can specify its session token id, instead of relying on our generating one. See TestSessionToken for example

> If each of 1 or 2 is not true we can just not call setSessionToken inside Client.Do but insted in Client.CreateTestRun

Yes, but it's not well design to just handle specific case for CreateTestRun. In future, if we introduce more Post endpoint which need idempotency, this design works for all of them.

Reply:
The problem is that ... literally we have 1 POST we need this for ... and while there is possibility we might add it to others, this is *not* the case today and I find it unlikely. 
Also the method name is ... lying it's not just setting a SessionToken, it's setting only if ... some conditions are met. 
To be honest I think `generateSessionToken` and `k6SessionTokenHeader` should become a public use them in `CreateTestRun` and let people if want to to set them in a custom request and send it ... which is also unlikely use case but still ... it would be possible.
If we ever want to do it for all Posts I would argue that the `if method == POST` should be in the body of the `client.Do` not in some function

Reply:
> If we ever want to do it for all Posts I would argue that the if method == POST should be in the body of the client.Do not in some function

Fair point, and with satori uuid, the code looks much cleaner